File: frontend/food-delivery-platform/src/models/requests/order/CreateOrderRequest.ts
import {CreateLocationRequest} from "./CreateLocationRequest";

export interface CreateOrderDishRequest {
    businessId : string
    orderedBy : string
    orderDate : string
    totalPrice : number
    deliveredBy? : string | null
    deliverTo : CreateLocationRequest
    deliverFrom : CreateLocationRequest
    paymentMethod : number
    dishes : CreateOrderDishRequest[]
}

File: backend/Contracts/DF.Contracts/Gateway/Requests/Order/CreateOrderRequest.cs
using System;
using System.Collections.Generic;
using DF.Contracts.Enums;

namespace DF.Contracts.Gateway.Requests.Order;

public record CreateOrderRequest(
    Guid BusinessId,
    Guid OrderedBy,
    DateTime OrderDate,
    decimal TotalPrice,
    Guid? DeliveredBy,
    CreateLocationRequest DeliverTo,
    CreateLocationRequest DeliverFrom,
    PaymentMethod PaymentMethod,
    List<CreateOrderDishRequest> Dishes
);